src/main/mondrian/rolap/SqlConstraintUtils.java

```diff
@@ -757,6 +757,22 @@ public static boolean hasMultiPositionSlicer(List<Member> slicerMembers) {
         return members;
     }
 
```

Comment:
The new expandSupportedCalculatedMember is basically the same as the block inside the loop of the original.  Would be nice to eliminate duplication.

One thing the new method omitted was handing disjoint tuples, however.  That may be something we need to disallow here as well-- not completely clear to me.  Have you tested that condition?

Also, this new method has an issue that bugged me with the original-- if a calculated member is unexpandable it will just return that calculated member.  So you may get back a list of base members, but you also might get back a calc member or empty list.  I wouldn't be surprised if there are some existing bugs lurking because that wasn't expected.  We should at least have some clear javadoc that says what this method will return given the various conditions.  

It seems like there should be a nice simple way to ask for a clean list of fully expanded base members, if possible, with some way of knowing that expansion was not fully possible if it fails.